src/CaptureClient/CaptureClient.cpp

```diff
@@ -154,10 +154,11 @@ ErrorMessageOr<CaptureListener::CaptureOutcome> CaptureClient::CaptureSync(
   CaptureOptions* capture_options = request.mutable_capture_options();
   capture_options->set_trace_context_switches(collect_scheduling_info);
   capture_options->set_pid(process_id);
-  if (samples_per_second == 0) {
+  if (samples_per_second == 0 || stack_dump_size == 0) {
```

Comment:
It could be that. But in that case, we would also need to tell our leaf function patching to not happen. Otherwise, all samples will be reported with "too small stack sample" error.

Reply:
Do you think it's worth doing that? Otherwise we could maybe add a `CHECK` that `stack_dump_size > 0` (not sure if this is the best place). I don't really like that this being 0 is a way of turning of sampling in addition to `samples_per_second`.

Reply:
I don't have string feelings here. I removed the condition here and added the CHECK to the point of reading the flag.